### ISSUE TITLE
Add `--icpc-strict` flag 

### ISSUE THREAD

**Issue by RagnarGrootKoerkamp** [RagnarGrootKoerkamp/BAPCtools#72]
Warn/error whenever non-icpc-spec features are used.

Add a `--icpc` command line arg and/or `icpc: true` to `contest.yaml` to validate that the problem adheres to the ICPC subset

**Comment by mzuenni** [comment]
is this still relevant for the draft PR?
- `answer_validators` are not in the spec but there existence should not break anything
- timelimit is handled spec conform
- visualizers are (will be) spec conform
- generators are spec conform
- ??

**Comment by RagnarGrootKoerkamp** [comment]
yeah; generally i think the whole concept of icpc subset doesn't have a lot of weight currently
